QWeb/internal/javascript.py

```diff
@@ -195,513 +111,100 @@ def get_all_elements(css: str) -> list[WebElement]:
-            const iterator = document.createNodeIterator(mainDoc, NodeFilter.SHOW_TEXT);
-            const textNodes = [];
-            let currentNode;
-            while ((currentNode = iterator.nextNode())) {
-                if(currentNode.textContent.trim() === text){
-                    textNodes.push(currentNode.parentElement);
-                    continue;
-                }
-                else if (partial && currentNode.textContent.trim().includes(text)) {
-                    textNodes.push(currentNode.parentElement);
-                }
-            }
-            return textNodes;
-        }
-        return(getTextNodes(arguments[0], arguments[1], arguments[2]))
-        """
+    """Find elements whose textContent matches to preferred text
+       (using external JS file, preloaded)."""
+    js = _FIND_TEXT_FROM_TEXTNODES_JS
     doc = "html"
     partial = kwargs.get("partial_match")
     return execute_javascript(js, text, doc, partial)
 
 
 def get_clickable(locator: str) -> list[WebElement]:
-    js = (
-        """
-    var web_elements = function(locator){
-        var full = [];
-        var partial = [];
-        var candidates = [];
-        var text = "";
-        var elems = document.querySelectorAll('button, a, label,\
-        *[type="submit"], *[type="button"], *[type="reset"], li[data-value], input[type="radio"],\
-        *[role="tab"], *[role="button"], *[ng-click], *[data-ng-click],[href]');
-        for (var i = 0; i < elems.length; i++) {
-            if(elems[i].tagName.toLowerCase() === 'input'){
-                text = elems[i].value;
-            }
-            else {
-                text = elems[i].innerText;
-            }
-            if (text.trim() === locator){
-                full.push(elems[i]);
-            }
-            else if (text.trim().toLowerCase() === locator.toLowerCase()){
-                partial.push(elems[i]);
-            }
-        }
-        candidates = full.concat(partial);
-        return candidates;
-    }
-    return(web_elements('"""
-        + locator.replace("'", "\\'")
-        + """'));"""
-    )
-    return execute_javascript(js)
+    """Find clickable elements matching the locator (using external JS file, preloaded)."""
+    js = _GET_CLICKABLE_JS
+    return execute_javascript(js, locator)
 
 
 def get_recursive_walk() -> str:
-    return """var recursiveWalk = function(node, func) {
-    var done = func(node);
-    if(done) {
-        return true;
-    }
-    if ('shadowRoot' in node && node.shadowRoot) {
-        var done = recursiveWalk(node.shadowRoot, func);
-        if (done) {
-            return true;
-        }
-    }
-    node = node.firstChild;
-
-    while (node) {
-        var done = recursiveWalk(node, func);
-        if (done) {
-            return true;
-        }
-        node = node.nextSibling;
-    }
-
-    }"""
+    """Return the recursiveWalk JS function as a string (using external JS file, preloaded)."""
+    return _GET_RECURSIVE_WALK_JS
```

Comment:
If this function is not called outside of this file I would remove it and just refer to the returned "constant" in the functions that call this one.

e.g.
`js = _GET_RECURSIVE_WALK_JS + "\n" + _GET_TEXT_ELEMENTS_FROM_SHADOW_DOM_JS`

At least for me, calling this function raises a suspicion that there's some additional logic / side-effect in it which I need to check to be sure.

Reply:
ok, will change in a new commit.

Reply:
should be changed now.